File: test_dashboard.java
package assesment;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class test_dashboard {
    public static void main(String[] args) {
        
        WebDriver driver = new ChromeDriver();

        try {
            driver.manage().window().maximize();
            driver.get("https://ua.segwise.ai/qa_assignment");

            
            WebDriverWait wait = new WebDriverWait(driver, 20);

            
            WebElement loginWithEmailBtn = wait.until(
                ExpectedConditions.elementToBeClickable(
                    By.xpath("//button//span[contains(text(),'Log in with email')]")
                )
            );
            loginWithEmailBtn.click();

            
            WebElement emailField = wait.until(
                ExpectedConditions.visibilityOfElementLocated(
                    By.cssSelector("input[autocomplete='email']")
                )
            );
            new Actions(driver).moveToElement(emailField).click().sendKeys("qa@segwise.ai").perform();

            
            WebElement passwordField = wait.until(
                ExpectedConditions.visibilityOfElementLocated(
                    By.cssSelector("input[type='password']")
                )
            );
            new Actions(driver).moveToElement(passwordField).click().sendKeys("segwise_test").perform();

            
            WebElement loginBtn = wait.until(
                ExpectedConditions.elementToBeClickable(By.xpath("//button[@type='submit']"))
            );
            loginBtn.click();

           
            wait.until(ExpectedConditions.urlContains("qa_assignment/home"));

            
            WebElement costChart = wait.until(
                ExpectedConditions.visibilityOfElementLocated(
                    By.xpath("//*[contains(text(),'Cost Per Install')]")
                )
            );

            if (costChart.isDisplayed()) {
                System.out.println("✅ Dashboard chart 'Cost Per Install' is visible!");
            } else {
                System.out.println("Chart not found!");
            }

        } catch (Exception e) {
            System.out.println("Test Failed: " + e.getMessage());
        } finally {
            try { Thread.sleep(5000); } catch (InterruptedException e) {}
            driver.quit();
        }
    }
}

File: test_dashboard.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

def test_dashboard():
    
    driver = webdriver.Chrome()

    try:
        driver.maximize_window()
        driver.get("https://ua.segwise.ai/qa_assignment")

        wait = WebDriverWait(driver, 20)

       
        login_with_email_btn = wait.until(
            EC.element_to_be_clickable((By.XPATH, "//button//span[contains(text(),'Log in with email')]"))
        )
        login_with_email_btn.click()

        #this is for email field
        email_field = wait.until(
            EC.visibility_of_element_located((By.CSS_SELECTOR, "input[autocomplete='email']"))
        )
        ActionChains(driver).move_to_element(email_field).click().send_keys("qa@segwise.ai").perform()

        
        password_field = wait.until(
            EC.visibility_of_element_located((By.CSS_SELECTOR, "input[type='password']"))
        )
        ActionChains(driver).move_to_element(password_field).click().send_keys("segwise_test").perform()

        
        login_btn = wait.until(
            EC.element_to_be_clickable((By.XPATH, "//button[@type='submit']"))
        )
        login_btn.click()

       
        wait.until(EC.url_contains("qa_assignment/home"))

        
        cost_chart = wait.until(
            EC.visibility_of_element_located((By.XPATH, "//*[contains(text(),'Cost Per Install')]"))
        )

        if cost_chart.is_displayed():
            print("Dashboard chart 'Cost Per Install' is visible!")
        else:
            print("Chart not found!")

    except Exception as e:
        print("Test Failed:", str(e))
    finally:
        time.sleep(5)
        driver.quit()

if __name__ == "__main__":
    test_dashboard()
